src/modules/mc_hover_thrust_estimator/zero_order_hover_thrust_ekf.cpp

```diff
@@ -124,24 +124,30 @@ inline void ZeroOrderHoverThrustEkf::updateStateCovariance(const float K, const
 	_state_var = math::constrain((1.f - K * H) * _state_var, 1e-10f, 1.f);
 }
 
-inline void ZeroOrderHoverThrustEkf::updateMeasurementNoise(const float residual, const float H)
+inline bool ZeroOrderHoverThrustEkf::isLargeOffsetDetected() const
 {
-	const float alpha = _dt / (noise_learning_time_constant + _dt);
-	const float res_no_bias = residual - _residual_lpf;
-	const float P = _state_var;
-	_acc_var = math::constrain((1.f - alpha) * _acc_var  + alpha * (res_no_bias * res_no_bias + H * P * H), 1e-4f, 60.f);
+	return fabsf(_signed_innov_test_ratio_lpf) > 0.2f;
 }
 
 inline void ZeroOrderHoverThrustEkf::bumpStateVariance()
 {
-	_state_var += 10000.f * _process_var * _dt;
+	_state_var += 1e3f * _process_var * _dt;
 }
 
-inline void ZeroOrderHoverThrustEkf::updateLpf(const float residual, const float innov_test_ratio)
+inline void ZeroOrderHoverThrustEkf::updateLpf(const float residual, const float signed_innov_test_ratio)
 {
-	const float alpha = _dt / (5.f * noise_learning_time_constant + _dt);
+	const float alpha = _dt / (_lpf_time_constant + _dt);
 	_residual_lpf = (1.f - alpha) * _residual_lpf + alpha * residual;
-	_innov_test_ratio_lpf = (1.f - alpha) * _innov_test_ratio_lpf + alpha * math::min(innov_test_ratio, 5.f);
+	_signed_innov_test_ratio_lpf = (1.f - alpha) * _signed_innov_test_ratio_lpf + alpha * math::constrain(
```

Comment:
I spent a lot of time working with a constrain-like function that had a signature of `limit(min, value, max)`, so now I have to double-check every time I see `constrain` used :laughing:

Reply:
Interesting I've never seen a different order... here's a good exaple of the same order: https://docs.scipy.org/doc/numpy/reference/generated/numpy.clip.html